generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -164,6 +166,175 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
+   * @param depth - how deep we are in the expression.
+   * @param fuzzer - the fuzzer object for generating fuzzed expressions.
+   * @param isZero - true if we are making an opaque zero, false otherwise.
+   * @return Optional.empty() if an opaque value can't be generated, otherwise an opaque value
+   *     made from performing a bitwise operation on an opaque zero or opaque one.
+   */
+  private Optional<Expr> opaqueZeroOrOneBitwiseOp(BasicType type, boolean constContext,
+                                                   final int depth, Fuzzer fuzzer, boolean isZero) {
+    if (!BasicType.allIntegerTypes().contains(type)) {
+      return Optional.empty();
+    }
+    if (!shadingLanguageVersion.supportedBitwiseOperations()) {
+      return Optional.empty();
+    }
+    final int operator = generator.nextInt(3);
+    Optional<Expr> opaqueExpr;
+    switch (operator) {
+      case 0:
+        // Bitwise AND
+        if (isZero) {
+          opaqueExpr = Optional.of(
+              new ParenExpr(
+                  generator.nextBoolean()
+                  ? new BinaryExpr(
+                      fuzzer.fuzzExpr(type, false, constContext, depth),
+                      makeOpaqueZero(type, constContext, depth, fuzzer),
+                      BinOp.BAND)
+                  : new BinaryExpr(
+                      makeOpaqueZero(type, constContext, depth, fuzzer),
+                      fuzzer.fuzzExpr(type, false, constContext, depth),
+                      BinOp.BAND)));
+        } else {
+          assert !isZero;
+          opaqueExpr = Optional.of(
+              new ParenExpr(
+                  new BinaryExpr(
+                      makeOpaqueOne(type, constContext, depth, fuzzer),
+                      makeOpaqueOne(type, constContext, depth, fuzzer),
+                      BinOp.BAND)));
+        }
+        break;
+      case 1:
+        // Bitwise OR
+        if (isZero) {
+          opaqueExpr = Optional.of(
+              new ParenExpr(
+                  new BinaryExpr(
+                      makeOpaqueZero(type, constContext, depth, fuzzer),
+                      makeOpaqueZero(type, constContext, depth, fuzzer),
+                      BinOp.BOR)));
+        } else {
+          assert !isZero;
+          opaqueExpr = Optional.of(
+              new ParenExpr(
+                  generator.nextBoolean()
+                  ? new BinaryExpr(
+                      makeOpaqueOne(type, constContext, depth, fuzzer),
+                      makeOpaqueZeroOrOne(generator.nextBoolean(), type, constContext, depth,
+                          fuzzer),
+                      BinOp.BOR)
+                  : new BinaryExpr(
+                      makeOpaqueZeroOrOne(generator.nextBoolean(), type, constContext, depth,
+                          fuzzer),
+                      makeOpaqueOne(type, constContext, depth, fuzzer),
+                      BinOp.BOR)));
+        }
+        break;
+      default:
+        // Bitwise XOR
```

Comment:
Change the '2' to (v - 1) where v is the variable you introduced to express the maximum number of options.